ktor-server/ktor-server-cio/common/src/io/ktor/server/cio/UnixSocketConnectorConfig.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2014-2025 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.cio
+
+import io.ktor.server.engine.ApplicationEngine
+import io.ktor.server.engine.ConnectorType
+import io.ktor.server.engine.EngineConnectorBuilder
+import io.ktor.server.engine.EngineConnectorConfig
+
+/**
+ * Represents a Unix domain socket connector configuration.
+ *
+ * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.server.engine.UnixSocketConnectorConfig)
+ */
+public interface UnixSocketConnectorConfig : EngineConnectorConfig {
```

Comment:
Is there a chance we can introduce a new interface, which will be a parent for `EngineConnectionConfig`? And then inherit from it. It seems possible to do it without introducing breaking changes, the only problem is naming.
It would also help us to add domain sockets support to other engines.

Reply:
Or at least, override `host` and `port` with deprecation in this interface.

Reply:
not in the minor release

Reply:
I think it's fine to have localhost and -1 as port as port is not defined

Reply:
What do you think about overriding with deprecation?